docs/development/cherry-picks.md

```diff
@@ -0,0 +1,44 @@
+# Overview
```

Comment:
Wouldn't https://github.com/gardener/gardener/blob/master/docs/development/process.md be a suitable place already for this documentation?

Reply:
No strong opinion, I now moved the docs to `./docs/development/process.md` as you suggested.